src/datastructure/RailwayNetwork.cpp

```diff
@@ -653,3 +657,749 @@ cda_rail::Network::Network(const std::filesystem::path &p) {
+        for (const auto& incoming_edge_index : in_edges(edge.target)) {
+            std::replace(successors[incoming_edge_index].begin(), successors[incoming_edge_index].end(), reverse_edge_index, new_reverse_edges.front());
+        }
+        for (size_t i = 0; i < new_reverse_edges.size() - 1; ++i) {
+            add_successor(new_reverse_edges[i], new_reverse_edges[i + 1]);
+        }
+    }
+
+    // Return
+    return return_edges;
+}
+
+std::pair<std::vector<int>, std::vector<int>> cda_rail::Network::separate_edge(int edge_index, cda_rail::SeparationType separation_type) {
+   /**
+    * Separates an edge (and possibly its reverse edge) according to the given number of new vertices.
+    *
+    * @param edge_index Index of the edge to separate.
+    * @param separation_type Type of separation.
+    */
+
+    // Check if graph is consistent for this function
+    if (!is_consistent_for_transformation()) {
+        throw std::invalid_argument("Graph is not consistent");
+    }
+
+    // Call function corresponding to separation type
+    if (separation_type == cda_rail::SeparationType::UNIFORM) {
+        return uniform_separate_edge(edge_index);
+    } else if (separation_type == cda_rail::SeparationType::CHEBYCHEV) {
+        return chebychev_separate_edge(edge_index);
+    } else {
+        throw std::invalid_argument("Separation type does not exist.");
+    }
+}
+
+std::pair<std::vector<int>, std::vector<int>> cda_rail::Network::uniform_separate_edge(int edge_index) {
+    /**
+     * Separates an edge (and possibly its reverse edge) uniformly according to the minimal block length.
+     */
+
+    // Check if graph is consistent for this function
+    if (!is_consistent_for_transformation()) {
+        throw std::invalid_argument("Graph is not consistent");
+    }
+
+    // Check if edge exists
+    if (!has_edge(edge_index)) {
+        throw std::invalid_argument("Edge does not exist");
+    }
+
+    // If the edge is not breakable throw an exception
+    if (!get_edge(edge_index).breakable) {
+        throw std::invalid_argument("Edge is not breakable");
+    }
+
+    // If the edge length is smaller than twice the minimum block length, nothing to separate
+    if (get_edge(edge_index).length < 2 * get_edge(edge_index).min_block_length) {
+        return std::make_pair(std::vector<int>(), std::vector<int>());
+    }
+
+    // Get edge to separate
+    const auto& edge = get_edge(edge_index);
+    // Get number of new vertices
+    double number_of_blocks = std::floor(edge.length / edge.min_block_length) ;
+
+    // Calculate distances
+    std::vector<double> distances_from_source;
+    for (int i = 1; i < number_of_blocks; ++i) {
+        distances_from_source.emplace_back(i * (edge.length / number_of_blocks));
+    }
+
+    return separate_edge_at(edge_index, distances_from_source);
+}
+
+std::pair<std::vector<int>, std::vector<int>> cda_rail::Network::chebychev_separate_edge(int edge_index) {
```

Comment:
It exists because I believe non-uniform block lengths might be superior, e.g., if they get smaller towards the edges that might be entrance points into stations. However, I removed this method until this is looked into and implemented. For now there is no reason for it to exist.